extensions/positron-assistant/src/participants.ts

```diff
@@ -461,6 +484,12 @@ abstract class PositronAssistantParticipant implements IPositronAssistantPartici
 			return;
 		}
 
+		// Track the model being used for this session
+		const toolContext = request.toolInvocationToken as any;
+		if (toolContext?.sessionId && request.model?.id) {
+			this._participantService.trackSessionModel(toolContext.sessionId, request.model.id);
+		}
+
```

Comment:
Nit: It would be simpler if we could handle this state in the same code path as `ChatRequestData`. Everywhere we call `getChatRequest` atm currently also has access to the `toolInvocationToken` and therefore probably also the session ID, so I think we could key both of these items on `sessionId`.

It's less important how exactly we track the state: in the service as you've done here versus in participants as we did for `ChatRequestData`.

Reply:
Makes sense!